src/Microsoft.Dnx.Runtime/Helpers/RoslynAnalyzerHelper.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.Dnx.Compilation;
+using Microsoft.Dnx.Compilation.CSharp;
+using NuGet;
+
+namespace Microsoft.Dnx.Runtime.Helpers
+{
+    internal static class RoslynAnalyzerHelper
+    {
+        private static readonly string AnalyzerBaseFolder = "analyzers";
+        private static readonly string AnalyzerLanguage = "cs";
+
+        public static IList<IAnalyzerReference> GetAnalyerReferences(PackageInfo packageInfo, string packagePath, FrameworkName runtimeFramework)
+        {
+            var files = new List<string>();
+
+            // assumption: analyzers' paths are in following format
+            // analyzer/{framework}/{cs|vb}/{filename}.dll
+            foreach (var relativePath in packageInfo.LockFileLibrary.Files)
+            {
+                if (!relativePath.StartsWith(AnalyzerBaseFolder, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                if (!relativePath.EndsWith(".dll", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                var parts = relativePath.Split(new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar });
+                if (parts.Length != 4)
+                {
+                    continue;
+                }
+
+                if (!string.Equals(parts[2], AnalyzerLanguage, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                var frameworkName = VersionUtility.ParseFrameworkFolderName(parts[1]);
+                if (VersionUtility.GetNearest(runtimeFramework, new[] { frameworkName }) != frameworkName)
```

Comment:
Regardless of how you fix it, add a unit test for this selection algorithm to ensure it selects the appropriate folders. It would also help discussion if you did that so I can point out concrete cases that might be missed :P.